pkg/sfu/downtrack.go

```diff
@@ -362,6 +370,64 @@ func (d *DownTrack) TrackInfoAvailable() {
 	d.connectionStats.Start(ti, time.Now())
 }
 
+func (d *DownTrack) SetStreamAllocatorListener(listener DownTrackStreamAllocatorListener) {
+	d.streamAllocatorLock.Lock()
+	d.streamAllocatorListener = listener
+	d.streamAllocatorLock.Unlock()
+}
```

Comment:
should `OnSubscriptionChanged` be called defensively here like it was in the callback binding function?

Reply:
Good call @paulwe . Will do that.